### ISSUE TITLE
A single bad spectrum in a .raw file cause all of Spectrum Searcher searching to fail 

### ISSUE THREAD

**Issue by benton-anderson** [benton-anderson/LipiDex-2#32]
Salma's raw file that clearly failed (it was 2 MB instead of 100MB+) had bad spectra that generated an error:

```
Date: 2023-06-22 12:07:18
Exception: Instrument index not available for requested device (Parameter 'instrumentIndex')
Stack Trace:    at ThermoFisher.CommonCore.RawFileReader.RawFileAccessBase.SelectInstrument(Device instrumentType, Int32 instrumentIndex)
   at Common.ThermoRawFileHandler..ctor(String filePath) in D:\LipidQC\LipidQC\Common\ThermoRawFileHandler.cs:line 32
   at SpectrumSearcher.Models.SpectrumSearcher.ReadThermoRawFile(String filepath) in D:\LipidQC\LipidQC\SpectrumSearcher\Models\SpectrumSearcher.cs:line 398
   at SpectrumSearcher.Models.SpectrumSearcher.ReadMassSpectrumFiles() in D:\LipidQC\LipidQC\SpectrumSearcher\Models\SpectrumSearcher.cs:line 339
   at SpectrumSearcher.Models.SpectrumSearcher.RunSpectrumSearcher(IProgress`1 uiProgressObject) in D:\LipidQC\LipidQC\SpectrumSearcher\Models\SpectrumSearcher.cs:line 107
   at SpectrumSearcher.ViewModels.SpectrumSearcherViewModel.<>c__DisplayClass115_0.<StartSpectrumSearch>b__2() in D:\LipidQC\LipidQC\SpectrumSearcher\ViewModels\SpectrumSearcherViewModel.Methods.cs:line 197

 ```

indicating that it couldn't get the instrument index. 

The solution should catch and log the bad spectrum or .raw file and then continue. 

Is Instrument index in one scan or is it in the header for the whole raw file?

**Comment by dbrademan** [comment]
I think best solution regarding bad raw files would be to run the relevant portion of Spectrum Searcher in a try-catch block, only catching the relevant errors. There are likely a couple. I'll check to see if there's a Raw File Accession superclass we can catch instead of this very specific exception.

**Comment by benton-anderson** [comment]
fixed in 0.1.4 with a try-catch block within the for loop
